src/mlpack/core/tree/rectangle_tree/rectangle_tree_impl.hpp

```diff
@@ -1409,33 +1409,33 @@ void RectangleTree<MetricType, StatisticType, MatType, SplitType, DescentType,
     parent = NULL;
   }
 
-  ar & BOOST_SERIALIZATION_NVP(maxNumChildren);
-  ar & BOOST_SERIALIZATION_NVP(minNumChildren);
-  ar & BOOST_SERIALIZATION_NVP(numChildren);
+  ar & CEREAL_NVP(maxNumChildren);
+  ar & CEREAL_NVP(minNumChildren);
+  ar & CEREAL_NVP(numChildren);
   if (Archive::is_loading::value)
     children.resize(maxNumChildren + 1);
 
-  ar & BOOST_SERIALIZATION_NVP(begin);
-  ar & BOOST_SERIALIZATION_NVP(count);
-  ar & BOOST_SERIALIZATION_NVP(numDescendants);
-  ar & BOOST_SERIALIZATION_NVP(maxLeafSize);
-  ar & BOOST_SERIALIZATION_NVP(minLeafSize);
-  ar & BOOST_SERIALIZATION_NVP(bound);
-  ar & BOOST_SERIALIZATION_NVP(stat);
-  ar & BOOST_SERIALIZATION_NVP(parentDistance);
-  ar & BOOST_SERIALIZATION_NVP(dataset);
-  ar & BOOST_SERIALIZATION_NVP(ownsDataset);
+  ar & CEREAL_NVP(begin);
+  ar & CEREAL_NVP(count);
+  ar & CEREAL_NVP(numDescendants);
+  ar & CEREAL_NVP(maxLeafSize);
+  ar & CEREAL_NVP(minLeafSize);
+  ar & CEREAL_NVP(bound);
+  ar & CEREAL_NVP(stat);
+  ar & CEREAL_NVP(parentDistance);
+  ar & CEREAL_POINTER(dataset);
+  ar & CEREAL_NVP(ownsDataset);
```

Comment:
I think this might need some different handling.  `ownsDataset` should always be true only if we are loading, and we should only serialize the dataset if we are the root node (i.e. `parent == NULL`).  Do you think you could apply the same strategies I've suggested for the other trees here?

Reply:
Yes.